### DISCUSSION TITLE
How to use Label as an argument for generic circuits?

### DISCUSSION THREAD

**Discussion by LeviPesin** [hneemann/Digital#1136] [Q&A] [answered]
Hello!

First of all, many thanks for the great program!

I would like to create a generic circuit (which sets a label of its internal component using `this.Label = args.Label` on it) which then can be used in another circuit, where the label can be set on this generic circuit and it would propagate to the internal component (to be used as a measurement value).
Is there a way to make the generic circuit infer `args.Label` to be just its label in the outer circuit -- so that writing `Label := "Some Label";` wouldn't be required in its generic parametrization (in the outer circuit)?

Also, why I can't do something like `this.isProgramCounter = args.Label == "PC"`? It throws `unexpected Token: EQUAL`...

**Comment by hneemann** [comment, marked as answer]
So far, it is not possible to access the parent label. But the idea is interesting. I will think about it.

> Also, why I can't do something like this.isProgramCounter = args.Label == "PC"? It throws unexpected Token: EQUAL...

Since '=' is not an expression, as e.g. in C, no '==' is required.
Whether an assignment or a comparison is meant can be concluded from the syntax.
